src/cli.js

```diff
@@ -218,17 +219,33 @@ async function runCmd (argv, stdout, stderr) {
       );
       if (existsSync(outDir))
         rimraf.sync(outDir);
+      mkdirp.sync(outDir);
       run = true;
 
     // fallthrough
     case "build":
       if (args._.length > 2)
         errTooManyArguments("build");
 
+      function hasTypeModule (path) {
+        let root = resolve('/');
+        while ((path = resolve(path, '..')) !== root) {
+          try {
+            return JSON.parse(readFileSync(eval('resolve')(path, 'package.json')).toString()).type === 'module';
+          }
+          catch (e) {
+            if (e.code === 'ENOENT')
+              continue;
+            throw e;
+          }
+        }
+      }
+
       let startTime = Date.now();
       let ps;
       const buildFile = eval("require.resolve")(resolve(args._[1] || "."));
-      const ext = buildFile.endsWith('.cjs') ? '.cjs' : '.js';
+      const esm = buildFile.endsWith('.mjs') || buildFile.endsWith('.js') && hasTypeModule(buildFile);
```

Comment:
There's a subtle difference in the implementation here and the `esm` default in in `src/index.js` file. Is there a reason for the difference?

Reply:
This line of `const esm = ` is the same for both, which lines do you mean?

Reply:
They're not the same

## src/cli.js

```js
const esm = buildFile.endsWith('.mjs') || buildFile.endsWith('.js') && hasTypeModule(buildFile);
```

## src/index.js

```js
esm = entry.endsWith('.mjs') || !entry.endsWith('.cjs') && hasTypeModule(entry),
```

## diff

```diff
- esm = entry.endsWith('.mjs') || entry.endsWith('.js') && hasTypeModule(entry)
+ esm = entry.endsWith('.mjs') || !entry.endsWith('.cjs') && hasTypeModule(entry)
```

Reply:
Thanks, yes these should be the same - I've pushed up a fix.